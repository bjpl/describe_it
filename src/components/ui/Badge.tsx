import React from "react";

interface BadgeProps {
  children: React.ReactNode;
  variant?: "default" | "secondary" | "outline" | "destructive";
  className?: string;
}

export const Badge: React.FC<BadgeProps> = ({
  children,
  variant = "default",
  className = "",
}) => {
  const variantStyles = {
    default: "bg-primary text-primary-foreground",
    secondary: "bg-secondary text-secondary-foreground",
    outline: "border-2 text-foreground",
    destructive: "bg-destructive text-destructive-foreground",
  };

  return (
    <span
      className={`inline-flex items-center rounded-full px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 ${variantStyles[variant]} ${className}`}
    >
      {children}
    </span>
  );
};

export default Badge;
