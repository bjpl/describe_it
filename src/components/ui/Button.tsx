import React from "react";

export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  variant?:
    | "default"
    | "secondary"
    | "destructive"
    | "outline"
    | "ghost"
    | "link";
  size?: "default" | "sm" | "lg" | "icon";
  children: React.ReactNode;
  loading?: boolean;
  leftIcon?: React.ReactNode;
  rightIcon?: React.ReactNode;
}

export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  (
    {
      className = "",
      variant = "default",
      size = "default",
      children,
      loading = false,
      leftIcon,
      rightIcon,
      disabled,
      ...props
    },
    ref,
  ) => {
    const variantStyles = {
      default: "bg-primary text-primary-foreground hover:bg-primary/90",
      secondary: "bg-secondary text-secondary-foreground hover:bg-secondary/80",
      destructive:
        "bg-destructive text-destructive-foreground hover:bg-destructive/90",
      outline:
        "border border-input bg-background hover:bg-accent hover:text-accent-foreground",
      ghost: "hover:bg-accent hover:text-accent-foreground",
      link: "text-primary underline-offset-4 hover:underline",
    };

    const sizeStyles = {
      default: "h-10 px-4 py-2",
      sm: "h-9 rounded-md px-3",
      lg: "h-11 rounded-md px-8",
      icon: "h-10 w-10",
    };

    const isDisabled = disabled || loading;

    return (
      <button
        className={`inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 ${variantStyles[variant]} ${sizeStyles[size]} ${className}`}
        ref={ref}
        disabled={isDisabled}
        aria-disabled={isDisabled}
        {...props}
      >
        {loading && (
          <svg
            className="animate-spin -ml-1 mr-2 h-4 w-4"
            xmlns="http://www.w3.org/2000/svg"
            fill="none"
            viewBox="0 0 24 24"
            aria-hidden="true"
          >
            <circle
              className="opacity-25"
              cx="12"
              cy="12"
              r="10"
              stroke="currentColor"
              strokeWidth="4"
            />
            <path
              className="opacity-75"
              fill="currentColor"
              d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
            />
          </svg>
        )}
        {!loading && leftIcon && (
          <span className="mr-2" aria-hidden="true">
            {leftIcon}
          </span>
        )}
        {children}
        {!loading && rightIcon && (
          <span className="ml-2" aria-hidden="true">
            {rightIcon}
          </span>
        )}
      </button>
    );
  },
);

Button.displayName = "Button";

export default Button;
