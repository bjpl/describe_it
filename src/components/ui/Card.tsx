import React from "react";

export interface CardProps {
  children: React.ReactNode;
  className?: string;
  clickable?: boolean;
  onClick?: () => void;
  role?: string;
  'aria-label'?: string;
  'aria-labelledby'?: string;
}

export interface CardHeaderProps {
  children: React.ReactNode;
  className?: string;
}

export interface CardTitleProps {
  children: React.ReactNode;
  className?: string;
  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
}

export interface CardDescriptionProps {
  children: React.ReactNode;
  className?: string;
}

export interface CardContentProps {
  children: React.ReactNode;
  className?: string;
}

export interface CardFooterProps {
  children: React.ReactNode;
  className?: string;
}

export const Card: React.FC<CardProps> = ({ 
  children, 
  className = "", 
  clickable = false,
  onClick,
  role,
  ...props 
}) => {
  const Component = clickable ? 'button' : 'div';
  
  return (
    <Component
      className={`rounded-lg border bg-card text-card-foreground shadow-sm ${
        clickable ? 'hover:shadow-md transition-shadow cursor-pointer focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2' : ''
      } ${className}`}
      onClick={clickable ? onClick : undefined}
      role={role}
      {...props}
    >
      {children}
    </Component>
  );
};

export const CardHeader: React.FC<CardHeaderProps> = ({
  children,
  className = "",
}) => {
  return (
    <div className={`flex flex-col space-y-1.5 p-6 ${className}`}>
      {children}
    </div>
  );
};

export const CardTitle: React.FC<CardTitleProps> = ({
  children,
  className = "",
  as: Component = 'h3',
}) => {
  return (
    <Component
      className={`text-2xl font-semibold leading-none tracking-tight ${className}`}
    >
      {children}
    </Component>
  );
};

export const CardDescription: React.FC<CardDescriptionProps> = ({
  children,
  className = "",
}) => {
  return (
    <p className={`text-sm text-muted-foreground ${className}`}>{children}</p>
  );
};

export const CardContent: React.FC<CardContentProps> = ({
  children,
  className = "",
}) => {
  return <div className={`p-6 pt-0 ${className}`}>{children}</div>;
};

export const CardFooter: React.FC<CardFooterProps> = ({
  children,
  className = "",
}) => {
  return (
    <div className={`flex items-center p-6 pt-0 ${className}`}>{children}</div>
  );
};
