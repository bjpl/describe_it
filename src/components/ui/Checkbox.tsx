"use client";

import React from "react";
import { cn } from "@/lib/utils";
import { Check, Minus } from "lucide-react";

export interface CheckboxProps {
  checked?: boolean;
  indeterminate?: boolean;
  onChange?: (checked: boolean) => void;
  onClick?: (event: React.MouseEvent) => void;
  disabled?: boolean;
  className?: string;
  size?: "sm" | "md" | "lg";
  "aria-label"?: string;
  "aria-labelledby"?: string;
  "aria-describedby"?: string;
}

export const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
  (
    {
      checked = false,
      indeterminate = false,
      onChange,
      onClick,
      disabled = false,
      className,
      size = "md",
      ...props
    },
    ref
  ) => {
    const sizeClasses = {
      sm: "h-4 w-4",
      md: "h-5 w-5",
      lg: "h-6 w-6",
    };

    const iconSizeClasses = {
      sm: "h-3 w-3",
      md: "h-4 w-4",
      lg: "h-5 w-5",
    };

    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
      onChange?.(event.target.checked);
    };

    const handleClick = (event: React.MouseEvent) => {
      onClick?.(event);
    };

    return (
      <div className="relative inline-flex items-center">
        <input
          ref={ref}
          type="checkbox"
          checked={checked}
          onChange={handleChange}
          onClick={handleClick}
          disabled={disabled}
          className="sr-only"
          {...props}
        />
        <div
          className={cn(
            "relative flex items-center justify-center border-2 rounded transition-all duration-200",
            sizeClasses[size],
            "focus-within:ring-2 focus-within:ring-ring focus-within:ring-offset-2",
            checked || indeterminate
              ? "bg-primary border-primary text-primary-foreground"
              : "border-input bg-background hover:border-primary/50",
            disabled && "opacity-50 cursor-not-allowed",
            !disabled && "cursor-pointer",
            className
          )}
          role="checkbox"
          aria-checked={indeterminate ? "mixed" : checked}
          tabIndex={disabled ? -1 : 0}
          onKeyDown={(e) => {
            if ((e.key === "Space" || e.key === "Enter") && !disabled) {
              e.preventDefault();
              onChange?.(!checked);
            }
          }}
        >
          {(checked || indeterminate) && (
            <div className="flex items-center justify-center">
              {indeterminate ? (
                <Minus className={cn(iconSizeClasses[size], "stroke-[3]")} />
              ) : (
                <Check className={cn(iconSizeClasses[size], "stroke-[3]")} />
              )}
            </div>
          )}
        </div>
      </div>
    );
  }
);

Checkbox.displayName = "Checkbox";

export default Checkbox;
