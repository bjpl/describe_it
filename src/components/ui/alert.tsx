/**
 * Alert UI Component
 * Displays important messages and notifications
 */

import React from 'react';
import { cn } from '@/lib/utils';
import { cva, type VariantProps } from 'class-variance-authority';

const alertVariants = cva(
  'relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-gray-950',
  {
    variants: {
      variant: {
        default: 'bg-white text-gray-950 border-gray-200',
        destructive:
          'border-red-200 text-red-800 bg-red-50 [&>svg]:text-red-600',
        warning:
          'border-yellow-200 text-yellow-800 bg-yellow-50 [&>svg]:text-yellow-600',
        success:
          'border-green-200 text-green-800 bg-green-50 [&>svg]:text-green-600',
        info:
          'border-blue-200 text-blue-800 bg-blue-50 [&>svg]:text-blue-600',
      },
    },
    defaultVariants: {
      variant: 'default',
    },
  }
);

interface AlertProps
  extends React.HTMLAttributes<HTMLDivElement>,
    VariantProps<typeof alertVariants> {}

const Alert = React.forwardRef<HTMLDivElement, AlertProps>(
  ({ className, variant, ...props }, ref) => (
    <div
      ref={ref}
      role="alert"
      className={cn(alertVariants({ variant }), className)}
      {...props}
    />
  )
);
Alert.displayName = 'Alert';

interface AlertTitleProps extends React.HTMLAttributes<HTMLHeadingElement> {}

const AlertTitle = React.forwardRef<HTMLParagraphElement, AlertTitleProps>(
  ({ className, ...props }, ref) => (
    <h5
      ref={ref}
      className={cn('mb-1 font-medium leading-none tracking-tight', className)}
      {...props}
    />
  )
);
AlertTitle.displayName = 'AlertTitle';

interface AlertDescriptionProps
  extends React.HTMLAttributes<HTMLParagraphElement> {}

const AlertDescription = React.forwardRef<
  HTMLParagraphElement,
  AlertDescriptionProps
>(({ className, ...props }, ref) => (
  <div
    ref={ref}
    className={cn('text-sm [&_p]:leading-relaxed', className)}
    {...props}
  />
));
AlertDescription.displayName = 'AlertDescription';

export { Alert, AlertTitle, AlertDescription };