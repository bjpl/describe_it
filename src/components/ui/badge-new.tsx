/**
 * Badge Component - Custom implementation to replace missing Radix UI component
 */
import React from 'react';
import { cn } from '@/lib/utils';

interface BadgeProps extends React.HTMLAttributes<HTMLSpanElement> {
  variant?: 'default' | 'secondary' | 'success' | 'warning' | 'error' | 'outline';
  size?: 'sm' | 'md' | 'lg';
}

const Badge: React.FC<BadgeProps> = ({ 
  className, 
  variant = 'default', 
  size = 'md',
  children,
  ...props 
}) => {
  const baseClasses = "inline-flex items-center font-semibold rounded-full border transition-colors";
  
  const variantClasses = {
    default: "bg-blue-100 text-blue-800 border-blue-200 dark:bg-blue-900 dark:text-blue-200",
    secondary: "bg-gray-100 text-gray-800 border-gray-200 dark:bg-gray-800 dark:text-gray-200",
    success: "bg-green-100 text-green-800 border-green-200 dark:bg-green-900 dark:text-green-200",
    warning: "bg-yellow-100 text-yellow-800 border-yellow-200 dark:bg-yellow-900 dark:text-yellow-200",
    error: "bg-red-100 text-red-800 border-red-200 dark:bg-red-900 dark:text-red-200",
    outline: "bg-transparent text-gray-800 border-gray-300 dark:text-gray-200 dark:border-gray-600",
  };
  
  const sizeClasses = {
    sm: "px-2 py-0.5 text-xs",
    md: "px-2.5 py-1 text-sm", 
    lg: "px-3 py-1.5 text-base",
  };

  return (
    <span
      className={cn(
        baseClasses,
        variantClasses[variant],
        sizeClasses[size],
        className
      )}
      {...props}
    >
      {children}
    </span>
  );
};

export { Badge };
export type { BadgeProps };