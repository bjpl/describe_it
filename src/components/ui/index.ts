// Core UI Components
export { Button } from "./Button";
export { Card, CardHeader, CardTitle, CardDescription, CardContent, CardFooter } from "./Card";
export { Input } from "./Input";
export { Progress } from "./Progress";
export { Badge } from "./Badge";
export { RadioGroup } from "./RadioGroup";
export { Select } from "./Select";
export { Tabs } from "./Tabs";

// Enhanced Components
export {
  Skeleton,
  EnhancedSkeleton,
  SkeletonGroup,
  TextSkeleton,
  ProfileSkeleton,
  CardSkeletonEnhanced,
  TableSkeleton,
} from "./Skeleton";

export type { Toast } from "./Toast";
export {
  ToastProvider,
  useToast,
  toast,
} from "./Toast";

export {
  Modal,
  ModalHeader,
  ModalTitle,
  ModalDescription,
  ModalContent,
  ModalFooter,
  ConfirmModal,
} from "./Modal";

export {
  Dropdown,
  SimpleDropdown,
} from "./Dropdown";

export {
  Tooltip,
  Popover,
} from "./Tooltip";

export {
  DataTable,
} from "./DataTable";

export {
  Checkbox,
} from "./Checkbox";

export {
  FormField,
  FormLabel,
  FormDescription,
  FormMessage,
  FormInput,
  FormTextarea,
  useFormValidation,
} from "./FormField";

// Motion Components - Export individual components from MotionComponents
export {
  MotionDiv,
  MotionButton,
  MotionSpan,
  MotionP,
  MotionH1,
  MotionH2,
  MotionH3,
  MotionH4,
  MotionH5,
  MotionH6,
  MotionForm,
  MotionInput,
  MotionTextarea,
  MotionSelect,
  MotionHeader,
  MotionSection,
  MotionArticle,
  MotionAside,
  MotionNav,
  MotionMain,
  MotionFooter,
  MotionA,
  MotionImg,
  MotionUl,
  MotionOl,
  MotionLi,
  MotionLink,
} from "./MotionComponents";

// Fallback components (if needed for compatibility)
export {
  MotionDiv as FallbackMotionDiv,
  MotionButton as FallbackMotionButton,
  MotionSpan as FallbackMotionSpan,
  FallbackAnimatePresence,
} from "./MotionFallback";

// Types - Export from their respective source files
export type { ButtonProps } from "./Button";
export type {
  CardProps,
} from "./Card";
export type { ModalProps } from "./Modal";
export type { TooltipProps, PopoverProps } from "./Tooltip";
export type { DropdownProps, DropdownItem } from "./Dropdown";
export type { DataTableProps, Column } from "./DataTable";
export type { CheckboxProps } from "./Checkbox";
export type {
  FormFieldProps,
  FormLabelProps,
  FormMessageProps,
  FormInputProps,
  FormTextareaProps,
  ValidationRule,
  UseFormValidationProps,
} from "./FormField";
