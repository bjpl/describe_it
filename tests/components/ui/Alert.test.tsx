import React from 'react';
import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import '@testing-library/jest-dom';
import { Alert, AlertTitle, AlertDescription } from '@/components/ui/alert';

describe('Alert Components', () => {
  describe('Alert Component', () => {
    it('should render children', () => {
      render(<Alert>Alert content</Alert>);
      expect(screen.getByText('Alert content')).toBeInTheDocument();
    });

    it('should have role="alert"', () => {
      render(<Alert>Alert</Alert>);
      expect(screen.getByRole('alert')).toBeInTheDocument();
    });

    it('should apply custom className', () => {
      const { container } = render(<Alert className="custom-alert">Alert</Alert>);
      expect(container.firstChild).toHaveClass('custom-alert');
    });

    it('should have default base styles', () => {
      const { container } = render(<Alert>Alert</Alert>);
      expect(container.firstChild).toHaveClass('relative', 'w-full', 'rounded-lg', 'border', 'p-4');
    });

    describe('Variants', () => {
      it('should render default variant', () => {
        const { container } = render(<Alert variant="default">Default</Alert>);
        expect(container.firstChild).toHaveClass('bg-white', 'text-gray-950', 'border-gray-200');
      });

      it('should render destructive variant', () => {
        const { container } = render(<Alert variant="destructive">Destructive</Alert>);
        expect(container.firstChild).toHaveClass('border-red-200', 'text-red-800', 'bg-red-50');
      });

      it('should render warning variant', () => {
        const { container } = render(<Alert variant="warning">Warning</Alert>);
        expect(container.firstChild).toHaveClass('border-yellow-200', 'text-yellow-800', 'bg-yellow-50');
      });

      it('should render success variant', () => {
        const { container } = render(<Alert variant="success">Success</Alert>);
        expect(container.firstChild).toHaveClass('border-green-200', 'text-green-800', 'bg-green-50');
      });

      it('should render info variant', () => {
        const { container } = render(<Alert variant="info">Info</Alert>);
        expect(container.firstChild).toHaveClass('border-blue-200', 'text-blue-800', 'bg-blue-50');
      });
    });

    describe('Ref Forwarding', () => {
      it('should forward ref correctly', () => {
        const ref = React.createRef<HTMLDivElement>();
        render(<Alert ref={ref}>Alert</Alert>);
        expect(ref.current).toBeInstanceOf(HTMLDivElement);
      });
    });

    describe('Display Name', () => {
      it('should have correct display name', () => {
        expect(Alert.displayName).toBe('Alert');
      });
    });

    describe('Icon Support', () => {
      it('should render with icon', () => {
        render(
          <Alert>
            <svg data-testid="alert-icon">Icon</svg>
            <div>Content</div>
          </Alert>
        );
        expect(screen.getByTestId('alert-icon')).toBeInTheDocument();
      });

      it('should apply icon styles via CSS', () => {
        const { container } = render(
          <Alert>
            <svg>Icon</svg>
            <div>Content</div>
          </Alert>
        );
        // Icon styles are applied via CSS classes
        expect(container.querySelector('svg')).toBeInTheDocument();
      });
    });
  });

  describe('AlertTitle Component', () => {
    it('should render children', () => {
      render(<AlertTitle>Alert Title</AlertTitle>);
      expect(screen.getByText('Alert Title')).toBeInTheDocument();
    });

    it('should render as h5 element', () => {
      const { container } = render(<AlertTitle>Title</AlertTitle>);
      expect(container.querySelector('h5')).toBeInTheDocument();
    });

    it('should apply custom className', () => {
      render(<AlertTitle className="custom-title">Title</AlertTitle>);
      expect(screen.getByText('Title')).toHaveClass('custom-title');
    });

    it('should have title styles', () => {
      render(<AlertTitle>Title</AlertTitle>);
      expect(screen.getByText('Title')).toHaveClass('mb-1', 'font-medium', 'leading-none', 'tracking-tight');
    });

    it('should forward ref correctly', () => {
      const ref = React.createRef<HTMLParagraphElement>();
      render(<AlertTitle ref={ref}>Title</AlertTitle>);
      expect(ref.current).toBeInstanceOf(HTMLHeadingElement);
    });

    it('should have correct display name', () => {
      expect(AlertTitle.displayName).toBe('AlertTitle');
    });
  });

  describe('AlertDescription Component', () => {
    it('should render children', () => {
      render(<AlertDescription>Alert description</AlertDescription>);
      expect(screen.getByText('Alert description')).toBeInTheDocument();
    });

    it('should render as div element', () => {
      const { container } = render(<AlertDescription>Description</AlertDescription>);
      expect(container.querySelector('div')).toBeInTheDocument();
    });

    it('should apply custom className', () => {
      const { container } = render(
        <AlertDescription className="custom-description">Description</AlertDescription>
      );
      expect(container.firstChild).toHaveClass('custom-description');
    });

    it('should have description styles', () => {
      const { container } = render(<AlertDescription>Description</AlertDescription>);
      expect(container.firstChild).toHaveClass('text-sm');
    });

    it('should forward ref correctly', () => {
      const ref = React.createRef<HTMLParagraphElement>();
      render(<AlertDescription ref={ref}>Description</AlertDescription>);
      expect(ref.current).toBeInstanceOf(HTMLDivElement);
    });

    it('should have correct display name', () => {
      expect(AlertDescription.displayName).toBe('AlertDescription');
    });
  });

  describe('Complete Alert Composition', () => {
    it('should render complete alert with all subcomponents', () => {
      render(
        <Alert variant="warning">
          <svg data-testid="warning-icon">âš </svg>
          <AlertTitle>Warning</AlertTitle>
          <AlertDescription>This is a warning message</AlertDescription>
        </Alert>
      );

      expect(screen.getByRole('alert')).toBeInTheDocument();
      expect(screen.getByTestId('warning-icon')).toBeInTheDocument();
      expect(screen.getByText('Warning')).toBeInTheDocument();
      expect(screen.getByText('This is a warning message')).toBeInTheDocument();
    });

    it('should work without title', () => {
      render(
        <Alert>
          <AlertDescription>Description only</AlertDescription>
        </Alert>
      );

      expect(screen.getByText('Description only')).toBeInTheDocument();
    });

    it('should work without description', () => {
      render(
        <Alert>
          <AlertTitle>Title only</AlertTitle>
        </Alert>
      );

      expect(screen.getByText('Title only')).toBeInTheDocument();
    });
  });

  describe('Accessibility', () => {
    it('should be accessible with title and description', () => {
      render(
        <Alert>
          <AlertTitle>Accessible Alert</AlertTitle>
          <AlertDescription>This alert is accessible</AlertDescription>
        </Alert>
      );

      const alert = screen.getByRole('alert');
      expect(alert).toBeInTheDocument();
    });

    it('should support custom HTML attributes', () => {
      render(
        <Alert data-testid="custom-alert" aria-label="Custom alert">
          Content
        </Alert>
      );

      expect(screen.getByTestId('custom-alert')).toBeInTheDocument();
      expect(screen.getByLabelText('Custom alert')).toBeInTheDocument();
    });
  });

  describe('Snapshot Tests', () => {
    it('should match snapshot for default alert', () => {
      const { container } = render(
        <Alert>
          <AlertTitle>Title</AlertTitle>
          <AlertDescription>Description</AlertDescription>
        </Alert>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for all variants', () => {
      const variants = ['default', 'destructive', 'warning', 'success', 'info'] as const;

      variants.forEach(variant => {
        const { container } = render(
          <Alert variant={variant}>
            <AlertTitle>{variant}</AlertTitle>
            <AlertDescription>{variant} description</AlertDescription>
          </Alert>
        );
        expect(container.firstChild).toMatchSnapshot();
      });
    });
  });
});
