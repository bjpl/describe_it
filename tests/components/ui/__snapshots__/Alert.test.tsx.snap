// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert Components > Snapshot Tests > should match snapshot for all variants 1`] = `
<div
  class="relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-gray-950 bg-white text-gray-950 border-gray-200"
  role="alert"
>
  <h5
    class="mb-1 font-medium leading-none tracking-tight"
  >
    default
  </h5>
  <div
    class="text-sm [&_p]:leading-relaxed"
  >
    default
     description
  </div>
</div>
`;

exports[`Alert Components > Snapshot Tests > should match snapshot for all variants 2`] = `
<div
  class="relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-gray-950 border-red-200 text-red-800 bg-red-50 [&>svg]:text-red-600"
  role="alert"
>
  <h5
    class="mb-1 font-medium leading-none tracking-tight"
  >
    destructive
  </h5>
  <div
    class="text-sm [&_p]:leading-relaxed"
  >
    destructive
     description
  </div>
</div>
`;

exports[`Alert Components > Snapshot Tests > should match snapshot for all variants 3`] = `
<div
  class="relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-gray-950 border-yellow-200 text-yellow-800 bg-yellow-50 [&>svg]:text-yellow-600"
  role="alert"
>
  <h5
    class="mb-1 font-medium leading-none tracking-tight"
  >
    warning
  </h5>
  <div
    class="text-sm [&_p]:leading-relaxed"
  >
    warning
     description
  </div>
</div>
`;

exports[`Alert Components > Snapshot Tests > should match snapshot for all variants 4`] = `
<div
  class="relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-gray-950 border-green-200 text-green-800 bg-green-50 [&>svg]:text-green-600"
  role="alert"
>
  <h5
    class="mb-1 font-medium leading-none tracking-tight"
  >
    success
  </h5>
  <div
    class="text-sm [&_p]:leading-relaxed"
  >
    success
     description
  </div>
</div>
`;

exports[`Alert Components > Snapshot Tests > should match snapshot for all variants 5`] = `
<div
  class="relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-gray-950 border-blue-200 text-blue-800 bg-blue-50 [&>svg]:text-blue-600"
  role="alert"
>
  <h5
    class="mb-1 font-medium leading-none tracking-tight"
  >
    info
  </h5>
  <div
    class="text-sm [&_p]:leading-relaxed"
  >
    info
     description
  </div>
</div>
`;

exports[`Alert Components > Snapshot Tests > should match snapshot for default alert 1`] = `
<div
  class="relative w-full rounded-lg border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-gray-950 bg-white text-gray-950 border-gray-200"
  role="alert"
>
  <h5
    class="mb-1 font-medium leading-none tracking-tight"
  >
    Title
  </h5>
  <div
    class="text-sm [&_p]:leading-relaxed"
  >
    Description
  </div>
</div>
`;
